spirv_glsl.cpp

```diff
@@ -3371,6 +3374,13 @@ string CompilerGLSL::access_chain_internal(uint32_t base, const uint32_t *indice
 				expr += ".";
 				expr += to_member_name(*type, index);
 			}
+
+			if (member_is_packed_type(*type, index))
+			{
+				auto &membertype = get<SPIRType>(type->member_types[index]);
+				expr = unpack_expression_type(expr, membertype);
```

Comment:
Yes. I wrestled with this...and actually explored passing the equivalent of need_transpose back...but ultimately aborted that out of concern about complexity.

As an l-value, there are two primary use cases for unpacking:
1) Allow assignment to a variable defined as the same vector type, but which would not be generally packed.
2) Allow component access using x, y... syntax instead of [0], [1]... Metal does not permit the former on packed types for some reason.
 
Regarding the potential use as an r-value...that is indeed problematic...but I am hoping very rare. This packing only applies to uniform buffer blocks...as input...and not to vertex or other array content that would be more commonly written to.